astropy/coordinates/representation.py

```diff
@@ -20,23 +20,29 @@
 from ..extern import six
 from ..utils import ShapedLikeNDArray, classproperty
 from ..utils.compat import NUMPY_LT_1_12
-from ..utils.compat.numpy import broadcast_arrays
-
-__all__ = ["BaseRepresentation", "CartesianRepresentation",
-           "SphericalRepresentation", "UnitSphericalRepresentation",
-           "PhysicsSphericalRepresentation", "CylindricalRepresentation"]
-
-# Module-level dict mapping representation string alias names to class.
-# This is populated by the metaclass init so all representation classes
-# get registered automatically.
+from ..utils.compat.numpy import broadcast_arrays, broadcast_to
+
+__all__ = ["RepresentationBase", "BaseRepresentation",
+           "CartesianRepresentation", "SphericalRepresentation",
+           "UnitSphericalRepresentation", "RadialRepresentation",
+           "PhysicsSphericalRepresentation", "CylindricalRepresentation",
+           "BaseDifferential", "CartesianDifferential",
+           "BaseSphericalDifferential", "BaseSphericalCosLatDifferential",
+           "SphericalDifferential", "SphericalCosLatDifferential",
+           "UnitSphericalDifferential", "UnitSphericalCosLatDifferential",
+           "RadialDifferential", "CylindricalDifferential",
+           "PhysicsSphericalDifferential"]
+
+# Module-level dict mapping representation string alias names to classes.
+# This is populated by the metaclass init so all representation and differential
+# classes get registered automatically.
 REPRESENTATION_CLASSES = {}
+DIFFERENTIAL_CLASSES = {}
```

Comment:
In the current formulation, these have to be separate because the dict's are keyed by the short name. i.e. `REPRESENTATION_CLASSES['spherical']` and `DIFFERENTIAL_CLASSES['spherical']` have the same key name, but return different classes.